bin/init.bash

```diff
@@ -6,6 +6,7 @@
 # It's not to be executed on its own but rather via `ck8s init`.
 
 : "${CK8S_CLUSTER:?Missing CK8S_CLUSTER}"
+: "${CLUSTER_PROVISIONER:?Missing CLUSTER_PROVISIONER}"
```

Comment:
Please add this as a variable comparable `CK8S_ENVIRONMENT_NAME`/`CK8S_FLAVOR`/`CK8S_CLOUD_PROVIDER` in common.bash:
```diff
diff --git a/bin/common.bash b/bin/common.bash
index 11c83ebd..5c6a862f 100644
--- a/bin/common.bash
+++ b/bin/common.bash
@@ -9,6 +9,7 @@
 # shellcheck disable=SC2034
 ck8s_cloud_providers=(
     "aws"
+    "azure"
     "baremetal"
     "citycloud"
     "exoscale"
@@ -18,6 +19,13 @@ ck8s_cloud_providers=(
     "none"
 )
 
+# shellcheck disable=SC2034
+ck8s_cluster_provisioner=(
+    "cluster-api"
+    "kubespray"
+    "other"
+)
+
 # shellcheck disable=SC2034
 ck8s_flavors=(
     "dev"
```

Then in init to support it:
```diff
diff --git a/bin/init.bash b/bin/init.bash
index e09d79b6..b98af2b4 100755
--- a/bin/init.bash
+++ b/bin/init.bash
@@ -18,6 +18,7 @@ source "${here}/common.bash"
 # Load cloud provider, environment name, and flavor from config if available.
 if [ -f "${config[default_common]}" ]; then
     cloud_provider=$(yq4 '.global.ck8sCloudProvider' "${config[default_common]}")
+    cluster_provisioner=$(yq4 '.global.ck8sClusterProvisioner' "${config[default_common]}")
     environment_name=$(yq4 '.global.ck8sEnvironmentName' "${config[default_common]}")
     flavor=$(yq4 '.global.ck8sFlavor' "${config[default_common]}")
 fi
@@ -29,6 +30,14 @@ elif [ -v CK8S_CLOUD_PROVIDER ] && [ "${CK8S_CLOUD_PROVIDER}" != "${cloud_provid
 else
     export CK8S_CLOUD_PROVIDER="${cloud_provider}"
 fi
+if [ -z "${cluster_provisioner:-}" ]; then
+    : "${CK8S_CLUSTER_PROVISIONER:?Missing CK8S_CLUSTER_PROVISIONER}"
+elif [ -v CK8S_CLUSTER_PROVISIONER ] && [ "${CK8S_CLUSTER_PROVISIONER}" != "${cluster_provisioner}" ]; then
+    log_error "ERROR: Cloud provider mismatch, '${cluster_provisioner}' in config and '${CK8S_CLUSTER_PROVISIONER}' in env"
+    exit 1
+else
+    export CK8S_CLOUD_PROVIDER="${cloud_provider}"
+fi
 if [ -z "${environment_name:-}" ]; then
     : "${CK8S_ENVIRONMENT_NAME:?Missing CK8S_ENVIRONMENT_NAME}"
 elif [ -v CK8S_ENVIRONMENT_NAME ] && [ "${CK8S_ENVIRONMENT_NAME}" != "${environment_name}" ]; then
@@ -410,16 +419,21 @@ set_cluster_dns() {
         exit 1
     fi
     case ${CLUSTER_PROVISIONER} in
-        CAPI)
+        cluster-api)
             clusterapi=true
             clusterdns="10.233.0.10"
             clusterdnscidr="10.233.0.10/32"
         ;;
-        *)
+        kubespray)
             clusterapi=false
             clusterdns="10.233.0.3"
             clusterdnscidr="10.233.0.3/32"
         ;;
+        *)
+            clusterapi=false
+            clusterdns="set-me"
+            clusterdnscidr="set-me"
+        ;;
     esac
 
     replace_set_me "${file}" ".global.clusterDns" "\"${clusterdns}\""
```

This will also require migration steps to set the correct one for current clusters.